src/base_home.h

```diff
@@ -0,0 +1,142 @@
+//NETWORKED -> Network enabled; able to form connection between command and remote system.
+
+/**
+ * Stores capacities of command system and the auxiliary systems present.
+ */
+struct command_sys
+{
+    bool network = false;  /**Is the command system wifi enabled? (cmd core must be networked)*/
+    bool auto_conn = true; /**Automaticaly connect to remote systems if able*/
+    int max_conn = 0;      /**Network capacity (in generic units).*/
+    int max_cpu;           /**Maximum processing power (in generic units.*/
+
+    std::unordered_map<furn_id, int> aux_sys;    /**Maps <auxiliary system type, # in command system>*/
+    std::unordered_map<furn_id, int> remote_sys; /**Maps <remote system type, # connected to command system*/
+};
+
+class base_home
+{
+
+    base(submap &base_map, const tripoint &coreloc);
+
+  private:
+    //########PRIVATE##########//
+    //#BASE CORE AND LOCATIONS#//
+    //#########################//
+
+    //data members
+    int base_level; /**Level of command core. Determins max # of aux sys and, use of certain features.*/
+    command_sys cmd_sys;
+
+    std::list<base_area_flag> baflag[SEEX][SEEY]; /**additional ter and fur flags specific to bases.*/
+    std::list<npc_based *> freeloaders;           /**list containing references to the NPCs staying at the base.*/
+
+    std::list<tripoint> bunks;        /**list containing locations of sleeping spots in base.*/
+    std::list<tripoint> storage_open; /**list containing locations of unclaimed/designated storage furnature.*/
+
+    //functions
+    void define_base_area(const submap &base_map, const tripoint &coreloc); //populates baflag[][]
+
+    //########PRIVATE##########//
+    //#ITEMS AND STASHED STUFF#//
+    //#########################//
+
+    //data members
+    std::unordered_map<ammotype_id, int> gun_count;  /**stores <ammo_type, #guns_that_use_them>.*/
+    std::unordered_map<ammotype_id, int> ammo_count; /**stores <ammo_types, amount>.*/
+
+    std::list<tripoint> storage_comm; /**list containing locations of communal storage.*/
+
+    //functions
+    void count_guns(); //update gun_count
+
+    //########PRIVATE##########//
+    //#RULES AND STUFF FOR NPC#//
+    //#########################//
+
+    //data members
+    int food_ration_lv; /**Severity of food rationing. 0 = no rationing. Max = 3.*/
+    int ammo_ration_lv; /**Severity of ammo rationing. 0 = no rationing. Max = 3.*/
+    int med_ration_lv;  /**Severity of medicine/first aid rationing. 0 = no rationing. Max = 3.*/
+
+    //functions
+
+  public:
+    //##########PUBLIC#########//
+    //#BASE CORE AND LOCATIONS#//
+    //#########################//
+
+    bool is_in_base(const tripoint &); //Returns true if inside defined base area.
+    void change_lv(int new_level);
+    void run_design();
+
+    inline int get_level()
+    {
+        return base_level;
+    }
+    inline int get_max_pop()
+    {
+        return max(bunks.length, storage_open.length) + npcs.length;
```

Comment:
Is this supposed to call `std::max`? If so, you have to add the `std::` prefix.

Also note: when you want to *call* functions, you **must** use brackets. `bunks.length` is **not** a function call. Yes, even if there are no parameters, the `()` are still required.

Reply:
function of this being first use of c++.  That's how you do it in java.

Reply:
When I learned Java (several versions ago) it required brackets for function calls as well. Accessing properties (called data members in C++) are done without them, but `bunks` is a `std::list` and `length` is a member *function* of it, not a data member.

Btw (I missed this previously): getter functions and generally all functions that don't change the object should be `const`, as in `int get_max() const { ... }`.